options/options.go

```diff
@@ -0,0 +1,61 @@
+package options
```

Comment:
This should not be in a separate package. If there is some import case where only types are needed then it should be types pkg under `build`. But I don't see how these cases should be possible as these are just the types that `build` pkg users for its functions and not for anything else.

Reply:
> Not sure how much this collides with #1296, but if it does we should get that in before moving a lot of things around.

That PR only touches 3 lines in `build/build.go`.

Reply:
Actually, the intention was to also move `toSolveOpts` and decouple it from the driver abstraction, which I'm planning to embark on next. The intention would be to make a cleaner abstractions we can continue to iterrate on, right now `build` package has lots of things leaky pieces. Eventully such `options` package could provide a more high-level `SovleOpts` replacement.

Reply:
In addition to https://github.com/docker/buildx/issues/1483#issuecomment-1362011227

I'm quite skeptical of the `toSolveOpt` abstraction that wouldn't just be a gimmick that only sets some of the properties, leaving the complicated ones to `build` pkg internals. There are lots of places where build invocations directly interact with driver interface (booting, caps detection, artifact loading, result chaining, multi-node result joining etc.). What we don't want is that there is an extra abstraction that Buildx itself doesn't really use and now when we need to implement some important user-facing features like https://github.com/docker/buildx/issues/1377 or https://github.com/docker/buildx/issues/1104 (that I think both change quite a lot how `build` pkg is called), we find out that this extra abstraction does not work in this cases and does not let us implement these features.

Assuming the abstraction that doesn't have the described maintenance issues exists and it is worth investing in writing one, it should happen within the `build` pkg. The problem isn't with splitting bigger files into smaller ones or changing function signatures where it makes sense, but introducing new pkg boundaries where the pkg is not usable on its own. If after the `build` pkg refactoring, we end up in the state where public functions(files) can be split in a way they become useful independently and without sacrificing future dev cost, then it is cheap to do the pkg split.